src/js/client/editprefset-productList-widget.js

```diff
@@ -0,0 +1,142 @@
+/*
+ * editprefset-productList-widget
+ *
+ * A component that renders a list of all the applications/products, as well
+ * as including Generic Settings at the top of the list. Typically used on
+ * the left hand side for navigation and adding new products to an NP Set.
+ */
+/* global lunr */
+"use strict";
+
+var gpii  = fluid.registerNamespace("gpii");
+fluid.registerNamespace("gpii.devpmt");
+fluid.registerNamespace("gpii.devpmt.productList");
+
+/**
+ *  Main infusion component for a filterable product list.
+ */
+fluid.defaults("gpii.devpmt.productListWidget", {
+    gradeNames: ["gpii.devpmt.viewComponent"],
+    model: {
+        productListFilter: "",
+        allSolutionsSorted: []
+    },
+    modelListeners: {
+        productListFilter: {
+            func: "{that}.filterProductList"
+        },
+        allSolutionsSorted: {
+            func: "{that}.updateLunrIndex"
+        }
+    },
+    selectors: {
+        productListFilterContainer: "#pmt-filter-container",
+        productListItems: ".pmt-sidebar-toc-product",
+        productListLinks: ".pmt-sidebar-product-link"
+    },
+    bindings: {
+    },
+    templates: {
+        initial: "editprefset-productList-widget"
+    },
+    components: {
+        filter: {
+            type: "gpii.devpmt.filterWidget",
+            createOnEvent: "onMarkupRendered",
+            container: "{that}.dom.productListFilterContainer",
+            options: {
+                selectors: {
+                    initial: "#pmt-common-filter-area"
+                },
+                model: {
+                    filterText: "{productListWidget}.model.productListFilter"
+                }
+            }
+        }
+    },
+    lunrIndex: null,
+    invokers: {
+        updateLunrIndex: {
+            funcName: "gpii.devpmt.productList.updateProductListLunrIndex",
+            args: ["{that}"]
+        },
+        filterProductList: {
+            funcName: "gpii.devpmt.productList.filterProductList",
+            args: ["{that}", "{that}.dom.productListItems", "{that}.model.productListFilter"]
+        },
+        // Used when clicking on the product, or otherwise activating it from the list.
+        selectProduct: {
+            funcName: "gpii.devpmt.productList.selectProduct",
+            args: ["{that}", "{gpii.devpmt.editPrefs}", "{arguments}.0"]
+        }
+    },
+    events: {
+        activateProduct: null
+    },
+    markupEventBindings: {
+        productListLinks: {
+            method: "click",
+            args: ["{that}.selectProduct"]
+        }
+    }
+});
+
+gpii.devpmt.productList.selectProduct = function (that, devpmt, event) {
+    var appid = event.currentTarget.dataset.appid;
+    that.events.activateProduct.fire(appid);
```

Comment:
Is "activateProduct" event still in use? I don't find listeners for it or dynamic component created on it.

Reply:
good catch, I've taken this out.